packages/coding-standard/docs/phpstan_rules.md

```diff
@@ -1810,6 +1810,89 @@ $dateTime = new Nette\Utils\DateTime('now');
 
 <br>
 
+## Prefer raw data in test\'s dataProvider method instead from setUp()
+
+- class: [`PreferredRawDataInTestDataProviderRule`](../src/Rules/PreferredRawDataInTestDataProviderRule.php)
+
+```yaml
+# phpstan.neon
+services:
+    -
+        class: Symplify\CodingStandard\Rules\PreferredRawDataInTestDataProviderRule
+        tags: [phpstan.rules.rule]
+```
+
+```php
+<?php
+
+final class UseDataFromSetupInTestDataProviderTest extends TestCase
+{
+    private $data;
+
+    protected function setUp()
+    {
+        $this->data = [[true]];
+    }
+
+    public function provideFoo()
+    {
+        //other statement here
+
+        $this->setUp();
```

Comment:
It should work without this call, as phpunit calls this method implicitly

Reply:
so, the purpose of this rule to collect data set in the setup() and check if collected data used in dataprovider?

Reply:
Purpose of this rule is to prevent from using anything configured in `setUp()` in `dataProvider()`

E.g.

```php
protected function setUp() // this method is called 2nd
{
    $this->whatever = true;
}

public function provideData() // this method is called 1st
{
    // $this->whatever is "null" here = bug
    yield [$this->whatever];
}
```

Reply:
ok, i will try.

Reply:
implemented with check variable is instance of ThisType